src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -257,9 +259,25 @@ private void EndCallCore()
                 HttpContext.Response.ConsolidateTrailers(this);
             }
 
+            LogCallEnd();
+
             _callComplete = true;
         }
 
+        private void LogCallEnd()
+        {
+            var currentActivity = Activity.Current;
```

Comment:
Activity for HTTP request is stopped when HtpClientHandler returns, and it depends on HttpCompletionOptions (i.e. when headers are received r when response is received and buffered).
And I thought per our email convresation that at the moment grpc call ends and trailer are received, http request is already finished. Did I get it wrong?

I.e. do we know for sure this is Activity created for HTTP outgoing request? Otherwise, this Activity is tracking something else (e.g. incoming request).

Reply:
This is the activity on the server. All this logic happens inside the request pipeline so the server activity should always be here.

Client is work in progress.

Reply:
oh, I see, thanks for the info